File: services/grpc-csharp/Domain/Entities/PersonEntity.cs
// ==============================================================================
// File: Domain/Entities/PersonEntity.cs
// Purpose: EF Core row for persons_csharp. Property names match SQL columns.
// SOLID: SRP — persistence model. Domain Person is mapped in the repository.
// ==============================================================================

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Pgvector;

namespace OmniTest.Polyglot.Nexus.Api.CSharp.Domain.Entities;

/// <summary>EF entity mapped to persons_csharp (canonical schema).</summary>
[Table("persons_csharp")]
public sealed class PersonEntity
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; }

    [Column("first_name")]
    public string FirstName { get; set; } = string.Empty;

    [Column("last_name")]
    public string LastName { get; set; } = string.Empty;

    [Column("age")]
    public int Age { get; set; }

    [Column("sex")]
    public string Sex { get; set; } = string.Empty;

    [Column("marital_status")]
    public string MaritalStatus { get; set; } = string.Empty;

    [Column("children_count")]
    public int ChildrenCount { get; set; }

    [Column("living_place")]
    public string LivingPlace { get; set; } = string.Empty;

    [Column("occupation")]
    public string Occupation { get; set; } = string.Empty;

    [Column("national_code")]
    public string NationalCode { get; set; } = string.Empty;

    [Column("embedding")]
    public Vector? Embedding { get; set; }

    [Column("has_passport")]
    public bool HasPassport { get; set; }
}

File: services/grpc-python/infrastructure/db/person_entity.py
"""
File: infrastructure/db/person_entity.py
Purpose: ORM mapping to persons_python. Not imported by presentation.
SOLID: SRP — table mapping only. Domain Person is a separate dataclass.
"""

import uuid

from pgvector.sqlalchemy import Vector
from sqlalchemy import Boolean, Column, Integer, String
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import declarative_base

Base = declarative_base()


class PersonEntity(Base):
    """Row shape of persons_python. Column names match 02_create_tables.sql."""

    __tablename__ = "persons_python"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    first_name = Column(String(100), nullable=False)
    last_name = Column(String(100), nullable=False)
    age = Column(Integer, nullable=False)
    sex = Column(String(20), nullable=False)
    marital_status = Column(String(20), nullable=False)
    children_count = Column(Integer, nullable=False)
    living_place = Column(String(50), nullable=False)
    occupation = Column(String(50), nullable=False)
    national_code = Column(String(10), nullable=False)
    embedding = Column(Vector(384))
    has_passport = Column(Boolean, nullable=False, default=False)
